services/time_rounding_service.rb

```diff
@@ -0,0 +1,34 @@
+class TimeRoundingService
```

Comment:
This code does not seem to be related to `Adds the message model`, which is the title of this PR. Seems more like it snuck into a commit. Sadly, I beleive it still has the bug regarding the last quarter hour.
I suspect I should close the other PR, as this code will now be in the codebase ?